### PR TITLE
Add dark mode support

### LINKED ISSUE #1: Dark Mode Support
**Is your feature request related to a problem? Please describe.**
I'm always frustrated when the current portfolio template lacks a dark mode option, which makes it difficult to use in low-light environments.

**Describe the solution you'd like**
Implement a dark mode option in the portfolio template, where users can toggle between light and dark themes. This would make the template more user-friendly, especially for those who prefer working in darker environments.

**Describe alternatives you've considered**
I've considered using third-party browser extensions to apply a dark theme to the template, but this is not as seamless or well-integrated as having native dark mode support.

**Additional context**
Dark mode has become a standard across many websites and applications, as it reduces eye strain and enhances accessibility for users in low-light settings. Adding dark mode would also increase the attractiveness of the template for developers who prefer dark-themed UIs.

### PR DESCRIPTION
**Pull request by NitanJana**
## Summary

This PR close #1  by adding dark mode support.

## Changes Made

- Add load script and theme toggle button.
- Added tailwind classes to components for dark mode support.

### CODE REVIEW

File: src/components/ThemeToggle.tsx
Comment (yaralahruthik) [inline src/components/ThemeToggle.tsx:1]: I like that you used Preat to get this done. But honestly, vanilla JS would have been better.

File: src/global.css [hunk 1]
```diff
@@ -1,8 +1,11 @@
...
-  --color-primary: 0deg 0% 0%;
-  --color-secondary: 0deg 0% 24%;
+  --color-primary: 0 0% 0%;
+  --color-secondary: 0 0% 24%;
+  --color-hover: 0 0% 97%;
 
-  --color-hover: 0deg 0% 97%;
+  --color-primary-dark: 221 39% 11%;
+  --color-secondary-dark: 220 25% 30%;
+  --color-hover-dark: 220 30% 15%;
```
Comment (yaralahruthik) [inline src/global.css:8]: There is better way to set dark colors.

**Comment by NitanJana** [comment]
Got it. Thanks for the review.

**Comment by NitanJana** [comment]
> Also, setting the CSS variables for dark mode can be done in a cleaner way.

Can you tell how I may improve this?

**Comment by yaralahruthik** [comment]
> > Also, setting the CSS variables for dark mode can be done in a cleaner way.
> 
> Could you please elaborate on how I may improve this?

I like this approach for Tailwind https://ui.shadcn.com/docs/installation/manual

**Pull request closed without merging**